ResearchUXFactory/SBATextFieldOptions.swift

```diff
@@ -0,0 +1,144 @@
+ 
+ @note If the "validationRegex" is defined, then the "invalidMessage" should also be defined.
+ 
+ */
+open class SBATextFieldOptions: SBADataObject, SBATextFieldRange {
+    
+    public dynamic var validationRegex: String?
+    
+    public dynamic var invalidMessage: String?
+    
+    public dynamic var maximumLength: Int = 0
+    
+    public dynamic var minimumLength: Int = 0
+    
+    public dynamic var autocapitalizationType: UITextAutocapitalizationType = .none
+    
+    public dynamic var keyboardType: UIKeyboardType = .default
+    
+    public dynamic var shouldConfirm: Bool = false
+    
+    override open func dictionaryRepresentationKeys() -> [String] {
+        return super.dictionaryRepresentationKeys().appending(contentsOf:[#keyPath(maximumLength),
+                                                                          #keyPath(minimumLength),
+                                                                          #keyPath(validationRegex),
+                                                                          #keyPath(invalidMessage),
+                                                                          #keyPath(autocapitalizationType),
+                                                                          #keyPath(keyboardType),
+                                                                          #keyPath(shouldConfirm)])
+    }
+    
+    override open func mapValue(_ value: Any?, forKey key: String, withClassType classType: String?) -> Any? {
+        if key == #keyPath(autocapitalizationType), let name = value as? String {
+            return UITextAutocapitalizationType(key: name).rawValue
+        }
+        else if key == #keyPath(keyboardType), let name = value as? String {
+            return UIKeyboardType(key: name).rawValue
+        }
+        return super.mapValue(value, forKey: key, withClassType: classType)
+    }
+    
+    override open func defaultValue(forKey key: String) -> Any? {
+        if key == #keyPath(validationRegex), self.minimumLength > 0, self.maximumLength > 0 {
+            return "[[:ascii:]]{\(self.minimumLength),\(self.maximumLength)}"
+        }
+        else if key == #keyPath(invalidMessage), self.minimumLength > 0, self.maximumLength > 0 {
+            return Localization.localizedStringWithFormatKey("SBA_REGISTRATION_INVALID_PASSWORD_LENGTH_%@_TO_%@",
+                                                             NSNumber(value: self.minimumLength),
+                                                             NSNumber(value: self.maximumLength))
+        }
+        else if key == #keyPath(shouldConfirm) {
+            return true
+        }
+        return super.defaultValue(forKey: key)
+    }
+    
+}
+
+public class SBAPasswordOptions: SBATextFieldOptions {
+    
+    /**
+     By default, the minimum password length is 4
+     */
+    public static let defaultMinLength: Int = 4
```

Comment:
the default when a new study is created on Bridge is a minimum password length of 8--maybe that should be the default here too? shrug, it can be changed in either place so I'm not sure it matters